tests/e2e/test_rosacli_machine_pool.go

```diff
@@ -1426,3 +1426,74 @@ var _ = Describe("Edit machinepool",
 					ContainSubstring("At least one machine pool able to run OCP workload is required"))
 			})
 	})
+
+var _ = Describe("Upgrade machinepool",
+	labels.Feature.Machinepool,
+	func() {
+		defer GinkgoRecover()
+		var (
+			clusterID                 string
+			rosaClient                *rosacli.Client
+			machinePoolUpgradeService rosacli.MachinePoolUpgradeService
+			clusterConfig             *config.ClusterConfig
+		)
+
+		BeforeEach(func() {
+			var err error
+
+			By("Get the cluster")
+			clusterID = config.GetClusterID()
+			Expect(clusterID).ToNot(Equal(""), "ClusterID is required. Please export CLUSTER_ID")
+
+			By("Retrieve cluster config")
+			clusterConfig, err = config.ParseClusterProfile()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			machinePoolUpgradeService = rosaClient.MachinePoolUpgrade
+
+			By("Skip testing if the cluster is not a Classic cluster")
```

Comment:
nit: Could we not do this check with the cluster profile? That way we could run the check before initializing the cluster to avoid some unnecessary work